code/modules/clothing/masks/gasmask.dm

```diff
@@ -18,25 +18,36 @@
 
 /obj/item/clothing/mask/gas/attackby(obj/item/W,mob/user)
 	..()
+	if(type != "/obj/item/clothing/mask/gas")
```

Comment:
This is also ugly bad non-OOP.
Make method canMakeHelmet() on /obj/item/clothing/mask/gas, have it return a bool, and use that instead.